### PR TITLE
Upgrade guide for Jekyll 3

### PR DESCRIPTION
**Pull request by chrisfinazzo**
Per #3324 and comments in #3602, docs changes related to the upgrade process for 3.0 can land here. In their current state, there is some duplication on the [Upgrading][] page and the information is (obviously) in need of an update.

[Upgrading]: http://jekyllrb.com/docs/upgrading/

### CODE REVIEW

**Comment by chrisfinazzo** [comment]
We're repeating ourselves when describing the behavior of Absolute v. Relative Permalinks. Does this still need to be prefaced with a "Warning" label?

**Comment by chrisfinazzo** [comment]
Sure, I'll update that.

See also - #3610 - I think that should cover most of `_docs`.

File: site/_docs/upgrading.md [hunk 1]
```diff
@@ -85,7 +87,7 @@ to one or more config files (comma-delimited, no spaces).
 
 * `--no-server`
 * `--no-auto`
```
Comment (parkr) [inline site/_docs/upgrading.md]: this is actually `--no-watch` not the line below

**Comment by chrisfinazzo** [comment]
👍

**Pull request merged**